src/SmoFacade/BackupFileTools.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace AgDatabaseMove.SmoFacade
+{
+  using System;
+  using System.Text.RegularExpressions;
+
+
+  public static class BackupFileTools
+  {
+    public enum BackupType
+    {
+      Full, // bak  / D
+      Diff, // diff / I
+      Log // trn  / L
+    }
+
+    public static bool IsUrl(string path)
+    {
+      return Regex.IsMatch(path, @"(http(|s):\/)(\/[a-z0-9\.\-]+)+\.([a-zA-Z]+)$");

```

Comment:
Are we not using `try { new Uri(path) } catch { ... }` here to avoid non-HTTP URLs or something?
I feel like maintaining a regex for this is more complicated than it needs to be.

Reply:
Good idea, but I don't think `new Uri()` is going to throw for many of the cases that we want to block.
https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netcore-3.1
[This documentation](https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder.scheme?view=netcore-3.1#remarks) says it, in fact, won't throw, but we could use the schema property to determine the type